src/Grid/Column/Collector/DataObject/FieldDefinitionCollector.php

```diff
@@ -0,0 +1,179 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\StudioBackendBundle\Grid\Column\Collector\DataObject;
+
+use Exception;
+use Pimcore\Bundle\StaticResolverBundle\Models\DataObject\ClassDefinitionResolverInterface;
+use Pimcore\Bundle\StaticResolverBundle\Models\DataObject\DataObjectServiceResolverInterface;
+use Pimcore\Bundle\StudioBackendBundle\Exception\Api\NotFoundException;
+use Pimcore\Bundle\StudioBackendBundle\Grid\Column\ClassIdInterface;
+use Pimcore\Bundle\StudioBackendBundle\Grid\Column\ColumnCollectorInterface;
+use Pimcore\Bundle\StudioBackendBundle\Grid\Column\FolderIdInterface;
+use Pimcore\Bundle\StudioBackendBundle\Grid\Column\UseClassIdTrait;
+use Pimcore\Bundle\StudioBackendBundle\Grid\Column\UseFolderIdTrait;
+use Pimcore\Bundle\StudioBackendBundle\Grid\Schema\ColumnConfiguration;
+use Pimcore\Bundle\StudioBackendBundle\Grid\Util\ColumnFieldDefinition;
+use Pimcore\Bundle\StudioBackendBundle\Util\Constant\ElementTypes;
+use Pimcore\Model\DataObject\ClassDefinition\Data;
+use Pimcore\Model\DataObject\ClassDefinition\Data\Localizedfields;
+use Pimcore\Model\DataObject\ClassDefinition\Data\Objectbricks;
+use Pimcore\Model\DataObject\ClassDefinition\Layout;
+
+/**
+ * @internal
+ */
+final class FieldDefinitionCollector implements ColumnCollectorInterface, ClassIdInterface, FolderIdInterface
+{
+    use UseClassIdTrait;
+    use UseFolderIdTrait;
+
+    /**
+     * @var ColumnFieldDefinition[]
+     */
+    private array $groupedDefinitions = [];
+
+    public function __construct(
+        private readonly ClassDefinitionResolverInterface $classDefinitionResolver,
+        private readonly DataObjectServiceResolverInterface $dataObjectServiceResolver,
+    ) {
+    }
+
+    public function getCollectorName(): string
+    {
+        return 'data-object-field-definition';
+    }
+
+    public function getColumnConfigurations(array $availableColumnDefinitions): array
+    {
+        try {
+            $classDefinition = $this->classDefinitionResolver->getById($this->getClassId());
+        } catch (Exception) {
+            $classDefinition = null;
+        }
+
+        if (!$classDefinition) {
+            throw new NotFoundException('Class definition', $this->getClassId());
```

Comment:
Can't we just throw the NotFoundException in the catch block above?

Reply:
Unfortunately not...getById can return null, otherwise we have to throw it twice.